### PR TITLE
Create migrate-connector-to-poetry.md

### PR DESCRIPTION
**Pull request by marcosmarxm**
A lot of community connector doesn't have poetry enabled yet. This is a common request we need to ask contributors to update.

@alafanechere @natikgadzhi any way to improve this, let me know.
Is there a way to get the latest baseImage easily?

### CODE REVIEW

File: docs/contributing-to-airbyte/migrate-connector-to-poetry.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  allowedHosts:
+    hosts:
+      - *
+  connectorBuildOptions:
+    baseImage: docker.io/airbyte/python-connector-base:1.2.0@sha256:c22a9d97464b69d6ef01898edf3f8612dc11614f05a84984451dde195f337db9
+  connectorSubtype: api
+  connectorType: source
+  dockerImageTag: 0.1.0
+```
+3. Create a file called `pyproject.toml` and transfer all lib used in `setup.py` to this file
```
Comment (natikgadzhi) [inline docs/contributing-to-airbyte/migrate-connector-to-poetry.md:16]: Can we put this into a two-column before/after table instead so it's easy to look at visually? Or tabs? 

If we use either of those, will it look good on mobile layout, and will it revert to vertical layout automatically?

File: docs/contributing-to-airbyte/migrate-connector-to-poetry.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+* `setup.py`
+```python
+from setuptools import find_packages, setup
+
+MAIN_REQUIREMENTS = [
+    "airbyte-cdk~=0.1",
```
Comment (natikgadzhi) [inline docs/contributing-to-airbyte/migrate-connector-to-poetry.md:23]: Let's call out that 0.1 here is an example. It's very dated, I want to make it clear.

File: docs/contributing-to-airbyte/migrate-connector-to-poetry.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+readme = "README.md"
+documentation = "https://docs.airbyte.com/integrations/sources/orb"
+homepage = "https://airbyte.com"
+repository = "https://github.com/airbytehq/airbyte"
+[[tool.poetry.packages]]
+include = "source_openweather"
+
+[tool.poetry.dependencies]
+python = "^3.9,<3.12"
+airbyte-cdk = "^0.74.0"
```
Comment (natikgadzhi) [inline docs/contributing-to-airbyte/migrate-connector-to-poetry.md:85]:
```suggestion
airbyte-cdk = "^0"
```

Additionally, you should call out that the version of CDK should be unpinned unless you intentionally want to pin a specific one for the connector. In the future, all low-code sources will drop their pyproject.toml files and always run on latest CDK.

File: docs/contributing-to-airbyte/migrate-connector-to-poetry.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+[tool.poetry.scripts]
+source-openweather = "source_openweather.run:run"
+
+[tool.poetry.group.dev.dependencies]
+pytest = "^6.2"
```
Comment (natikgadzhi) [inline docs/contributing-to-airbyte/migrate-connector-to-poetry.md:91]:
```suggestion
pytest = "^8"
```

6.2 is ancient history, test deps should be updated.

File: docs/contributing-to-airbyte/migrate-connector-to-poetry.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+requests-mock = "^1.11.0"
```
Comment (natikgadzhi) [inline docs/contributing-to-airbyte/migrate-connector-to-poetry.md:92]:
```suggestion
requests-mock = "*"
```

File: docs/contributing-to-airbyte/migrate-connector-to-poetry.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+pytest-mock = "^3.6.1"
```
Comment (natikgadzhi) [inline docs/contributing-to-airbyte/migrate-connector-to-poetry.md:93]:
```suggestion
pytest-mock = "*"
```

**Review by natikgadzhi — APPROVED** [review]
Approving as is, with caveat that you accept suggestions.

There's a script that @alafanechere made that does poetry migration and nukes dockerfile (which your guide covers, but manually). We should mention that script, and I think make a loom with an example on one of the connectors. 

I've promised that guide to @imrehg — so I'm very interested in getting that in. 

A few things to add: 
- [ ] Add strong recommendation to use `airbyte-ci connectors --name source-whatever migrate_to_base_image` and `migrate_to_poetry`. Don't quote me on command names, check them. 
- [ ] Clarify that the same can be done manually with your example.
- [ ] Clarify that there needs to be a changelog line in README.md.
- [ ] Add a screencast with the whole process.

Are you intentionally not adding the page to docs sidebar? 

Let me know if you would like to merge this as is or want me to add the video overview and check the boxes — I'm happy to help, sort of planned on doing this anyway this week.

File: docs/contributing-to-airbyte/migrate-connector-to-poetry.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
+# Migrate connector to use Poetry and base Docker Image
+
+1. Remove the `Dockerfile` and `.dockerignore`
+2. Include the `baseImage` to the `metadata.yaml` file. For now, look for `baseImage` to find examples:
```
Comment (alafanechere) [inline docs/contributing-to-airbyte/migrate-connector-to-poetry.md:4]:
```suggestion
2. Include the `baseImage` to the `metadata.yaml` file. Please use the [latest non RC](https://hub.docker.com/r/airbyte/python-connector-base/tags) `airbyte/python-connector-base` image 
``` and add the image digest identifier (`@shar456:<digest>`)

File: docs/contributing-to-airbyte/migrate-connector-to-poetry.md (hunk 1 above)
Comment (alafanechere) [inline docs/contributing-to-airbyte/migrate-connector-to-poetry.md:16]: Users could also try `airbyte-ci --name=<connector-name> migrate_to_poetry`
It automatically does what you described below.

File: docs/contributing-to-airbyte/migrate-connector-to-poetry.md (hunk 7 above)
Comment (alafanechere) [inline docs/contributing-to-airbyte/migrate-connector-to-poetry.md:4]: Users can use `airbyte-ci connectors --name=<connector-name> migrate_to_base_image`

File: docs/contributing-to-airbyte/migrate-connector-to-poetry.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+[tool.poetry.scripts]
+source-openweather = "source_openweather.run:run"
+
+[tool.poetry.group.dev.dependencies]
+pytest = "^6.2"
+requests-mock = "^1.11.0"
+pytest-mock = "^3.6.1"
+```
+4. Delete the `setup.py` file
+5. Go to the connector folder and run `poetry lock` to generate the poetry file
```
Comment (alafanechere) [inline docs/contributing-to-airbyte/migrate-connector-to-poetry.md:96]: The can also run `poetry check` to validate the package is correctly declared

**Review by alafanechere — APPROVED** [review]
Feel free to mention `airbyte-ci` commands that can help automating the migration.

**Comment by natikgadzhi** [comment]
Oh gawd, we didn't merge this? 

@marcosmarxm at this point, we should not add this, most connectors are already on poetry with only 5 remaining. Closing.

**Pull request closed without merging**